src/analyze/analyze-security.c

```diff
@@ -225,7 +225,7 @@ static int assess_user(
         if (streq_ptr(info->user, NOBODY_USER_NAME)) {
                 d = "Service runs under as '" NOBODY_USER_NAME "' user, which should not be used for services";
                 b = 9;
-        } else if (info->dynamic_user && !STR_IN_SET(info->user, "0", "root")) {
+        } else if (info->dynamic_user && info->user && !STR_IN_SET(info->user, "0", "root")) {
```

Comment:
Hmm, if dynamic user is used, `User` will be automatically derived from unit name if not explicitly set. So there's no need for this check?

Reply:
ok